src/log.ts

```diff
@@ -90,38 +113,33 @@ export class Log {
         }
         const url = cluster.server + path;
 
-        const requestOptions: request.Options = {
-            method: 'GET',
-            qs: {
-                ...options,
-                container: containerName,
-            },
-            uri: url,
-        };
-        await this.config.applyToRequest(requestOptions);
-
-        return new Promise((resolve, reject) => {
-            const req = request(requestOptions, (error, response, body) => {
-                if (error) {
-                    reject(error);
-                    done(error);
-                } else if (response.statusCode !== 200) {
-                    try {
-                        const deserializedBody = ObjectSerializer.deserialize(JSON.parse(body), 'V1Status');
-                        reject(new HttpError(response, deserializedBody, response.statusCode));
-                    } catch (e) {
-                        reject(new HttpError(response, body, response.statusCode));
+        const requestOptions: RequestOptions = {};
+
+        const requestURL = new URL(cluster.server + path);
+
+        var searchParams = requestURL.searchParams;
+        AddOptionsToSearchParams(options, searchParams);
+
+        await this.config.applytoHTTPSOptions(requestOptions);
+        console.log(requestURL.toString())
+        const req = await fetch(requestURL.toString(), requestOptions)
+            .then(response => {
+                const status = response.status;
+                if (status == 200) {
```

Comment:
probably not strictly necessary, but triple equals here?

Reply:
got it